### ISSUE TITLE
mvn build should fail in case of an exception

### ISSUE THREAD

**Issue by wattazoum** [bsorrentino/maven-confluence-plugin#110]
Hi, 

I got a format exception (which I can correct) but the build ended successful.

```
Page: Runner - file:/home/wattazoum/Dev/workspace/git/greenpepper/greenpepper-maven-plugin/src/site/markdown/runner-ref.md
[ERROR] error loading template
org.codehaus.swizzle.confluence.ConfluenceException: java.lang.Exception: com.atlassian.confluence.rpc.RemoteException: com.atlassian.confluence.content.render.xhtml.migration.exceptions.UnknownMacroMigrationException: The macro 'classpaths' is unknown.
        at org.codehaus.swizzle.confluence.Confluence.call(Confluence.java:971)
        at org.codehaus.swizzle.confluence.Confluence.storePage(Confluence.java:359)
        at org.bsc.maven.confluence.plugin.AbstractConfluenceMojo.generateChild(AbstractConfluenceMojo.java:290)
        at org.bsc.maven.confluence.plugin.AbstractConfluenceSiteMojo.generateChildren(AbstractConfluenceSiteMojo.java:151)
        at org.bsc.maven.confluence.plugin.AbstractConfluenceSiteMojo.generateChildren(AbstractConfluenceSiteMojo.java:155)
        at org.bsc.maven.confluence.plugin.ConfluenceDeployMojo$2.execute(ConfluenceDeployMojo.java:708)
        at org.bsc.maven.confluence.plugin.AbstractBaseConfluenceMojo.confluenceExecute(AbstractBaseConfluenceMojo.java:189)
        at org.bsc.maven.confluence.plugin.ConfluenceDeployMojo.generatePluginReport(ConfluenceDeployMojo.java:677)
        at org.bsc.maven.confluence.plugin.ConfluenceDeployMojo.execute(ConfluenceDeployMojo.java:264)
        at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo(DefaultBuildPluginManager.java:134)
        at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:208)
        at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:153)
        at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:145)
        at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:116)
        at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:80)
        at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build(SingleThreadedBuilder.java:51)
        at org.apache.maven.lifecycle.internal.LifecycleStarter.execute(LifecycleStarter.java:128)
        at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:307)
        at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:193)
        at org.apache.maven.DefaultMaven.execute(DefaultMaven.java:106)
        at org.apache.maven.cli.MavenCli.execute(MavenCli.java:862)
        at org.apache.maven.cli.MavenCli.doMain(MavenCli.java:286)
        at org.apache.maven.cli.MavenCli.main(MavenCli.java:197)
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:498)
        at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced(Launcher.java:289)
        at org.codehaus.plexus.classworlds.launcher.Launcher.launch(Launcher.java:229)
        at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode(Launcher.java:415)
        at org.codehaus.plexus.classworlds.launcher.Launcher.main(Launcher.java:356)
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 16.967 s
[INFO] Finished at: 2016-08-04T23:56:50+02:00
[INFO] Final Memory: 21M/240M
[INFO] ------------------------------------------------------------------------
```

**Comment by bsorrentino** [comment]
The build should fail 

Let me check

**Comment by bsorrentino** [comment]
Hi

Currently the plugin doesn't break build if  publication, _concerning children_, fails 

Probably i can add a new configuration to force failure in such case

what do you think about ?

**Note:**
Please move to  version` 4.10` so it will be simpler for me manage the issue

**Comment by wattazoum** [comment]
I moved to version 4.10. 

Don't you think this should be the default behavior ? 
I believe most of people would want to be notified (by build failure) if part of their website is not updated.

**Comment by bsorrentino** [comment]
fix released in `5.0-rc1`
